processor/batchprocessor/factory.go

```diff
@@ -28,8 +28,9 @@ const (
 	// The value of "type" key in configuration.
 	typeStr = "batch"
 
-	defaultSendBatchSize = uint32(8192)
-	defaultTimeout       = 200 * time.Millisecond
+	defaultSendBatchSize            = uint32(8192)
+	defaultTimeout                  = 200 * time.Millisecond
+	defaultMetadataCardinalityLimit = 100
```

Comment:
I can see users asking already for guidance on setting this value. Is there a reason why this is 100 and not 1000? I have a _feeling_ that 100 is too low, but I don't know the impact of having 1000 as the default. Have you played with the settings, and can you write a couple of sentences on picking the right value?

Reply:
I am not sure what kind of user will think 100 is too low. In the scenario that Lightstep cares about, the number of metadata combinations is small, it's the number of projects per customer. The customer might have multiple projects in use, but they'll know how many that is. I wrote:
```
	// defaultMetadataCardinalityLimit should be set to the number
	// of metadata configurations the user expects to submit to
	// the collector.
```

Reply:
My view needs to be adjusted to the reality of common users of this, but when I think about metadata combinations, I think of tenant-id and cluster-name, for instance. In that case, 5 clusters and 20 tenants would already be 100 potential combinations. Going back to what I had originally in mind: is there guidance we can provide on the costs of 1000 vs. 100? Is it mostly about memory? There might also be a processing/scheduling penalty, but I believe this will be mostly memory, right?